### ISSUE TITLE
Accessibility for TooltipDefinition

### ISSUE THREAD

**Issue by gracelo** [IBM/carbon-components-react#1283]
## Accessibility for TooltipDefinition

> Describe in detail the issue you're having.
My react component has code like:
```
            <div ...className='parameter-list--nested-thread'>
              <TooltipDefinition onMouseOver={this.onMouseOver} tooltipText={name || 'undefined'} aria-label='tooltip for parameter'>
              </TooltipDefinition>
            </div>...
```

The generated html is like:
```
<div class="parameter-list--nested-thread">
  <div class="bx--tooltip--definition">
    <button class="bx--tooltip__trigger" aria-describedby="definition-tooltip-13"></button>
    <div id="definition-tooltip-13" class="bx--tooltip--definition__bottom" role="tooltip" aria-label="classes" style="top: 3px;">
       <span class="bx--tooltip__caret"></span><p>classes</p>
    </div>
  </div>
</div>
```
The DAP accessibility tool complains:
> Each form control must have associated label.
on the` <button> `tag line.

Please see if a label can be generated for it.

> Is this issue related to a specific component?
TooltipDefinition

> What did you expect to happen? What happened instead? What would you like to see changed?
no DAP error for the component

> What browser are you working in?
Chrome

> What version of the Carbon Design System are you using?
"carbon-components": "~9.1.13",
"carbon-components-react": "^6.12.1",

> What offering/product do you work on? Any pressing ship or release dates we should be aware of?
console
![image](https://user-images.githubusercontent.com/25369764/45229000-b8c9c380-b292-11e8-8ab5-6e4c8e7da51d.png)

**Comment by joshblack** [comment]
Hi @gracelo! 👋 Thanks so much for taking the time to make this issue 😄 

With this DAP error specifically, you need to have content in the Tooltip for the violation to pass. You can see in our [Storybook](http://react.carbondesignsystem.com/?knob-The%20title%20%28title%29=Section%201%20title&knob-Tooltip%20direction%20%28direction%29=bottom&knob-Tooltip%20content%20%28tooltipText%29=Brief%20description%20of%20the%20dotted%2C%20underlined%20word%20above.&selectedKind=TooltipDefinition&selectedStory=default&full=1&addons=1&stories=1&panelRight=0&addonPanel=storybook%2Factions%2Factions-panel) how this component is intended to be used, namely:

```jsx
<TooltipDefinition tooltipText="Brief description of the dotted, underlined word above.">
  Definition Tooltip
</TooltipDefinition>
```

This text is then placed inside the `button` which should get rid of this violation.

**Comment by gracelo** [comment]
However, in my UI design, the tooltip only shows the tooltipText on hovering.  We don't want to show any default text inside the TooltipDefinition.

**Comment by joshblack** [comment]
Sounds good! Could you share an image or gif that shows the behavior you're looking for in your UI?

**Comment by gracelo** [comment]
![image](https://user-images.githubusercontent.com/25369764/45230861-40fe9780-b298-11e8-83a8-cf548077a77f.png)

So the "vertical line" has the tooltip when hovering on, shows type name.

**Comment by joshblack** [comment]
Ah, unfortunately that's not an intended use-case for `TooltipDefinition`. Here's some information on the component:

![screen shot 2018-09-07 at 11 26 54 am](https://user-images.githubusercontent.com/3901764/45231230-f7f71500-b290-11e8-97cc-f38e8f05148f.png)

**Comment by gracelo** [comment]
Ok.  I will circle back to the team and may be that part cannot be accessible at the moment.

**Comment by tw15egan** [comment]
http://carbondesignsystem.com/components/tooltip/usage

**Comment by joshblack** [comment]
Going to close this for now, feel free to comment if there is still an issue!
